app/forms/manage_users_form.rb

```diff
@@ -0,0 +1,37 @@
+class ManageUsersForm
+  include ActiveModel::Model
+
+  attr_reader :team, :user_emails
+
+  validate :user_emails_are_valid_email_addresses
+
+  def initialize(team)
+    @team = team
+    @user_emails = team.users.map(&:email)
+  end
+
+  def submit(user_emails:)
+    @user_emails = user_emails
+
+    return false unless valid?
+
+    result = team.update(users: users_from_user_emails)
+    errors.merge!(team.errors)
+    result
+  end
+
+  private
+
+  def users_from_user_emails
+    user_emails.map do |email|
+      User.find_by(email: email).presence || User.new(email: email)
```

Comment:
Do you think it would be worth trying to combine this into one DB query instead of many?

```ruby
existing_users = User.where(email: user_emails).to_a
missing_users = (user_emails - existing_users.map(&:emails)).map { |e| User.new(email: e) }

(existing_users + missing_users).sort_by(&:email)
```

Reply:
Nice, I like it